sdk-lib/src/main/java/cash/z/ecc/android/sdk/block/processor/CompactBlockProcessor.kt

```diff
@@ -964,11 +964,17 @@ class CompactBlockProcessor internal constructor(
                         Twig.error {
                             "Downloading UTXO from height: $startHeight failed with: ${response.description}."
                         }
-                        throw LightWalletException.FetchUtxosException(
-                            response.code,
-                            response.description,
-                            response.toThrowable()
-                        )
+                        if (response is Response.Failure.Server.Unavailable) {
+                            Twig.error { "Download UTXOs failed - setting Disconnected state" }
+                            setState(State.Disconnected)
```

Comment:
Should this happen for every lightwalletd server request? If so, it seems error-prone to duplicate the logic at every site of a request.